test/unittests/OpModel/TestOpModelLib.cpp

```diff
@@ -0,0 +1,353 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "gtest/gtest.h"
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/MLIRContext.h"
+#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNN.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNNOpsAttrs.h"
+#include "ttmlir/OpModel/TTNN/TTNNOpModel.h"
+#include <llvm/ADT/ArrayRef.h>
+#include <llvm/Support/raw_ostream.h>
+#include <mlir/IR/BuiltinAttributeInterfaces.h>
+#include <mlir/IR/BuiltinTypes.h>
+
+#include "SingletonDeviceContext.h"
+
+namespace mlir::tt::op_model::ttnn {
+
+class OpModelBase : public ::testing::Test {
+public:
+  mlir::MLIRContext context;
+  mlir::OpBuilder builder = mlir::OpBuilder(&context);
+
+  void SetUp() override {
+    llvm::errs() << "MARKOBE SetUp\b";
+    context.loadDialect<mlir::tt::ttnn::TTNNDialect>();
+
+    setenv("TT_METAL_HOME",
+           "/proj_sw/user_dev/mbezulj/work/tt-mlir/third_party/tt-metal/src/"
+           "tt-metal/",
+           1);
+    SingletonDeviceContext::getInstance();
+  }
+  void TearDown() override { llvm::errs() << "MARKOBE TearDown\n"; }
+
+  mlir::tt::ttnn::TTNNLayoutAttr
+  CreateLayout(llvm::ArrayRef<int64_t> tensorShape,
+               mlir::tt::ttnn::BufferType bufferType,
+               mlir::tt::ttnn::TensorMemoryLayout tensorMemoryLayout,
+               ArrayRef<int64_t> gridShape = {8, 8}) {
+    return mlir::tt::ttnn::TTNNLayoutAttr::get(
+        &context, tensorShape, TileType::get(&context, builder.getBF16Type()),
+        bufferType, GridAttr::get(&context, gridShape),
+        mlir::tt::ttnn::TensorMemoryLayoutAttr::get(&context,
+                                                    tensorMemoryLayout));
+  }
+};
+
+template <typename T>
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
+                              const llvm::ArrayRef<T> &array) {
+  os << "[";
+  for (size_t i = 0; i < array.size(); ++i) {
+    if (i != 0)
+      os << ", ";
+    os << array[i];
+  }
+  os << "]";
+  return os;
+}
+
+TEST_F(OpModelBase, ReluInterleaved) {
```

Comment:
Can we also have tests that call the high level interface using RankedTensorTypes? Or is this also blocked by llvm lit problem?

Reply:
We can do that as well. And it would make sense to test interface on the op level as well. Though in that case we are doubling amount of test and maintenance / therefore I would prefer to use std::vector<std::tuple<shape, layout>> instead of RankedTensorType; or just remove these tests